packages/app/src/app/pages/common/Modals/MidTrialModal/index.tsx

```diff
@@ -58,48 +56,70 @@ export const MidTrialModal = () => {
     modalClosed();
   };
 
-  const handleUpgrade = () => {
+  const handleUpgrade = async () => {
     track('Mid trial reminder: upgrade to pro', {
       codesandbox: 'V1',
       event_source: 'UI',
     });
 
-    dismissMidTrialReminder();
+    // Custom implementation based on useGetCheckoutURL hook because the modal
+    // needs to be dismissed before redirecting.
+    try {
+      setLoadingPortal(true);
+      const payload = await api.stripeCustomerPortal(activeTeam, pathname);
```

Comment:
Can you replace this with the `useCreateCustomerPortal` hook?

Reply:
I shall take a look, thanks!

Reply:
@tristandubbeld we could pass an optional "onComplete" callback to the hook but if you think it's out of scope, we can merge this as is.